### PR TITLE
Update LM documentation

### PR DESCRIPTION
**Pull request by julianaklulo**
#### What
Update the LM documentation to include the new OLicense integration.
Also add information to the getting started page and refactor the other pages.

#### Why
To keep the documentatino in syc with the new features.

`Task`: https://jira.scania.com/browse/ASP-2029

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: docs-source/src/architecture.rst [hunk 1]
```diff
@@ -1,12 +1,12 @@
 License Manager Architecture
 ============================
-The license-manager is based on a client/backend architecture. The backend consists of a python http API and a
-postgresql database. The client consists of a timed reconcile job that runs on the control node in the cluster and
-a prolog integration to SLURM.
+The License Manager is based on a client/backend architecture. The backend consists of a Python HTTP API and a
```
Comment (matheushent) [inline docs-source/src/architecture.rst]:
```suggestion
The License Manager is based on a client/backend architecture. The backend consists of an HTTP API built of top of FastAPI and a
```

**Review by matheushent — COMMENTED** [review]
question: is it really needed to commit the binary files from `docs/.docstrees`?

**Comment by matheushent** [comment]
> question: is it really needed to commit the binary files from `docs/.docstrees`?

I think I understood what's going on. You commited both the source of the docs and the built docs. What I did in other repositories is to only commit the source of the docs, then provide instructions for building, e.g. [jobbergate docs](https://github.com/omnivector-solutions/jobbergate/blob/main/jobbergate-docs/.gitignore#L3).

File: docs-source/src/architecture.rst [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+PostgreSQL database. The client consists of a timed reconcile job that runs on the control node in the cluster and
```
Comment (dusktreader) [inline docs-source/src/architecture.rst]:
```suggestion
The License Manager is based on a client/backend architecture. The backend consists of a RESTful api built with Python over a
PostgreSQL database. The client consists of a timed reconcile job that runs on the control node in the cluster and
```
 [](edited-with-Axolo)

File: docs-source/src/getting_started.rst [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+You need to have the license server installed and working on a path that is accessible to the ``license-manager-agent``.
+The path for each license server binary is configurable in the ``license-manager-agent`` charm.
+In case you don't have a license server, you can use the ``license-manager-simulator`` to simulate the output of a license server.
+
+License Manager Backend
+***********************
+The ``license-manager-backend`` is an API that stores license usage information gathered from the license servers by the agent's reconciliation
+process. This data is used to update the license counters in the cluster to reflect the actual usage of the licenses.
+For each license tracked by License Manager, you need to create a configuration in the API, which includes the license name, the license
+features, the license server type and location, and the grace time (how long it takes for the license to be checked out after the job starts).
```
Comment (dusktreader) [inline docs-source/src/getting_started.rst]:
```suggestion
For each license tracked by License Manager, you need to create a configuration in the API. This includes the license name, the license
features, the license server type and location, and the grace time (how long it takes for the license to be checked out after the job starts).
```
 [](edited-with-Axolo)

**Review by dusktreader — APPROVED** [review]
Looks good! Added a couple of small suggestions, but it's good to go.

File: docs-source/src/architecture.rst (hunk 2 above)
Comment (julianaklulo) [reply docs-source/src/architecture.rst]: Thanks, I'll change it.

File: docs-source/src/getting_started.rst (hunk 3 above)
Comment (julianaklulo) [reply docs-source/src/getting_started.rst]: I'll change it, thanks!

File: docs-source/src/architecture.rst (hunk 1 above)
Comment (julianaklulo) [reply docs-source/src/architecture.rst]: I went with Tucker's suggestion for this line, thanks!

**Comment by julianaklulo** [comment]
> question: is it really needed to commit the binary files from `docs/.docstrees`?
>
> I think I understood what's going on. You commited both the source of the docs and the built docs. What I did in other repositories is to only commit the source of the docs, then provide instructions for building, e.g. [jobbergate docs](https://github.com/omnivector-solutions/jobbergate/blob/main/jobbergate-docs/.gitignore#L3).



@matheushent we have a subapp for LM docs source, and one with the docs built, which is deployed by GitHub Pages. I'm not sure if there's a way of making it automatically build and deploy, that's why we've been building it manually.
 [](edited-with-Axolo)

**Pull request merged**